src/hotspot/share/classfile/systemDictionaryShared.cpp

```diff
@@ -174,7 +174,6 @@ InstanceKlass* SystemDictionaryShared::acquire_class_for_current_thread(
 
   // No longer holding SharedDictionary_lock
   // No need to lock, as <ik> can be held only by a single thread.
-  loader_data->add_class(ik);
```

Comment:
Are the comments pertaining to the `add_class` call?

Reply:
Why not just move this to after the successful `load_shared_class` below? It is very hard to see the complete call sequence to when `restore_unshareable_info` gets called, to know that moving it there makes sense.

Reply:
The comment is about all that could happen to `ik` after the current thread have gotten exclusive ownership of `ik`.  I.e., no other thread will modify the contents of `ik` or store `ik` into a `ClassLoaderData`.

The `loader_data->add_class(ik)` for all other shared classes are done inside `restore_unshareable_info()`, so we should do the same thing in this case. This means the same sequence of actions will happen (e.g.,  `loader_data->add_class(ik)` happens before the the Java mirror is created.